tmp.yml

```diff
@@ -0,0 +1,66 @@
+# Also, TechPreview featureset is required here
+# https://docs.openshift.com/container-platform/4.15/nodes/clusters/nodes-cluster-enabling-features.html#nodes-cluster-enabling-features-cli_nodes-cluster-enabling
+
+# The nodeSelector is evaluated during pod scheduling only, that means that if you change the node labels after the pod was created it won't move.
+# For your test case this means that you have to label the node before changing the network.config settings
+
+# If no match node, the the pod will be in pending status, right ?
+# Pending means it is waiting for an available node
+```
+oc get pods  -n openshift-network-diagnostics  -o wide
+NAME                                    READY   STATUS    RESTARTS   AGE   IP            NODE                                        NOMINATED NODE   READINESS GATES
+network-check-source-75b88967df-mlfsn   0/1     Pending   0          43s   <none>        <none>  
+```
+
+```
+    - lastTransitionTime: "2024-04-22T09:07:59Z"
+      message: No source pods available
+      reason: NoSources
+      status: "False"
+      type: NetworkDiagnosticsAvailable
+```
+
+# mode: ""   means mode: All  ?
```

Comment:
https://github.com/openshift/api/pull/1800/files#diff-27a079803ccb4eb5829f5893ed44d6682a3dd99cfae4ffb3d74d35a8885c395dR223-R224

"" currently means `All` but in principal it means "sane default, the user doesn't care"